connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java

```diff
@@ -322,22 +337,18 @@ private boolean sendRecords() {
                             @Override
                             public void onCompletion(RecordMetadata recordMetadata, Exception e) {
                                 if (e != null) {
-                                    // Given the default settings for zero data loss, this should basically never happen --
-                                    // between "infinite" retries, indefinite blocking on full buffers, and "infinite" request
-                                    // timeouts, callbacks with exceptions should never be invoked in practice. If the
-                                    // user overrode these settings, the best we can do is notify them of the failure via
-                                    // logging.
```

Comment:
might want to edit these comments as well: https://github.com/apache/kafka/blob/2.3.0/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java#L548-L549

Reply:
Don't the changes in this PR make the comment you linked accurate, in the sense that those producer configs help prevent data loss? As long as the connector fails when the producer send fails, I think that comment is fine, and ultimately the theme of "don't change these if you don't want data loss" still rings true regardless.

Reply:
In the light of your findings, we should update that comment that "these settings will execute infinite retries when there are retriable exceptions, and overriding them...".

Reply:
SGTM, can't hurt to add a little more detail there.